### PR TITLE
feat: implement Hetzner Cloud support for virtual (shared) IP

### PR DESCRIPTION
**Pull request by sergelogvinov**
# Pull Request

Talos supports automatic virtual IP for the control plane with pure
layer 2 connectivity. Hetzner Cloud API supports assigning Floating IPs
to the nodes, this PR combines existing virtual IP functionality with calls
to HCloud API to move the IP address on HCloud side to the leader node.

The only thing which should be supplied in the machine configuration is
the Hetzner Cloud API token, every other setting is automatically
discovered by Talos.

Talos supports two types of floating IPs:
* external Floating IP for external network
* server alias IP for local networks

The controlplane can have only one alias on the local network interface.

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4255)

### CODE REVIEW

**Comment by sergelogvinov** [comment]
part of #3599

**Comment by smira** [comment]
probably missing `make generate`

File: internal/app/machined/pkg/controllers/network/operator/vip/hcloud.go [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+		}
+	}
+
+	return fmt.Errorf("error assigning %q to server %d: floating IP is not found", handler.vip, handler.deviceID)
+}
+
+// Release implements Handler interface.
+func (handler *HCloudHandler) Release(ctx context.Context) error {
+	if handler.networkID > 0 {
+		aliace := hcloud.ServerChangeAliasIPsOpts{
```
[the remark this answers is not in the document]
Comment (smira) [reply internal/app/machined/pkg/controllers/network/operator/vip/hcloud.go:120]: should be `alias` ? `aliasOptions` ?

[the remark this answers is not in the document]
Comment (sergelogvinov) [reply internal/app/machined/pkg/controllers/network/operator/vip/hcloud.go]: thanks, fixed

**Review by smira — APPROVED** [review]
thank you!

**Pull request merged**